Objetivo1

pedrooot opened the pull request:
### Sobre los milestones
* [x] ¿Las historias de usuario y los milestones constituyen una guía
    suficiente para poder comenzar el desarrollo de un proyecto?
* [x] ¿Los milestones están ordenados correctamente, de forma que el 1 o
   0 sea lo primero que hay que empezar a publicar?
* [x] ¿Todos los milestones se construyen sobre el anterior, teniendo un
   orden lógico el desarrollo?
* [ ] ¿El milestone 0 tiene asignada alguna historia de usuario?
* [ ] ¿El milestone 0 corresponde con lo mencionado varias veces en
   clase, y en concreto con lo que se solicita en el objetivo
   siguiente?
* [x] ¿Todos los milestones describen correctamente un producto, y no un
   concepto vago, una funcionalidad o una tarea?
* [x] ¿Los milestones son graduales, o hay un salto grande entre ellos?
Son graduales.
* [x] ¿Los milestones iniciales son internos, y hay a continuación milestones o
   productos externos?
* [x]  ¿Los milestones son mínimos, es decir, incluyen un producto
   mínimamente viable para poder avanzar?
* [ ]  ¿El milestone 0 es mínimo *y* viable?
* [x]  ¿Se indica claramente cómo comprobar la viabilidad del PMV de cada hito? Es
    decir, ¿se dice qué requisitos técnicos tiene que cumplir el producto de
    cada hito?
* [x] ¿Entre dos milestones consecutivos, no hay ningún PMV posible o por el
    contrario podrían desarrollarse muchas posibles etapas internas o externas
    de un proyecto?
Podrían desarrollarse en diferentes etapas.
### Sobre las historias de usuarios
* [x] ¿He tenido en cuenta en las historias de usuario que se trata de un
   proyecto que se desplegará en la nube?
* [x] ¿Están descritos todos los conceptos mencionados en las HUs con
   suficiente claridad, generalmente en documentos aparte?
* [x] ¿Tienen coherencia suficiente las historias de usuario, y en caso
   de que no lo tengan, se ha escrito un *user journey* para
   aclararlo?
* [x] ¿Todas las historias de usuario representan un beneficio para el
    mismo y se relacionan con la lógica de negocio? Es decir, ¿los usuarios
    *desean* hacerlo o tú deseas que el usuario te lo pida para hacerlo?

pedrooot committed:
primer commit objetivo 1

pedrooot committed:
Aniadi milestone

pedrooot committed:
aniade milestones que faltaban

pedrooot committed:
actualizo issues

jmramirezG commented:
Hola Pedro! Soy José María, me han asignado como revisor a tu proyecto.

En general las historias de usuario las veo bien.
Es cierto que sólo se da el punto de vista de tres personas que quieren llegar de A a B rápido, quizás sobran dos de esas 3 HU y se podría incluir alguna HU de un(a) concejal(a) de urbanismo que necesite usar la aplicación cómo se indica en la [lógica de negocio de tu readme](https://github.com/pedromarting3/OnTime#l%C3%B3gica-de-negocio).

Los milestones si que habría que tocar alguna cosilla, pero van bien encaminados:

* 2 de los 3 milestones no tienen ningún issue/HU asociados, intenta enfocarlos de forma que todos solucionen algún problema de los descritos en las HU.
* La forma de escribir el [M3](https://github.com/pedromarting3/OnTime/milestone/3) no representa realmente un PMV, quizás cambiar las palabras para que incluya algún producto? (un modelo, una función...).
* Hay que incluir cúando se da por resuelto un milestone, es decir, qué tiene que cumplir para que el hito descrito se solucione, algo parecido a lo que has hecho en el [M1](https://github.com/pedromarting3/OnTime/milestone/1) (*Las ciudades se representarán en forma de árbol en el que cada nodo representará una zona de la ciudad y su estado.*), pero también en el resto. Lo que si, intenta explicitar más cúando quedaría resuelto, por ejemplo, cambiando el M1 quedaría *Este milestone estará resuelto cuando podamos representar ciudades en forma de árbol, donde cada nodo corresponde a X y tiene un estado que indica Z, y cada arista corresponde a Y*.

Además, hay que incluir en la carpeta de la documentación las historias de usuario y los milestones, con un enlace a los mismos.

Cualquier duda puedes preguntarme sin problema y te la intentaré resolver, ánimo!!

JJ commented:
> Hola Pedro! Soy José María, me han asignado como revisor a tu proyecto.
> 
> En general las historias de usuario las veo bien. Es cierto que sólo se da el punto de vista de tres personas que quieren llegar de A a B rápido, quizás sobran dos de esas 3 HU y se podría incluir alguna HU de un(a) concejal(a) de urbanismo que necesite usar la aplicación cómo se indica en la [lógica de negocio de tu readme](https://github.com/pedromarting3/OnTime#l%C3%B3gica-de-negocio).
> 
> Los milestones si que habría que tocar alguna cosilla, pero van bien encaminados:
> 
>     * 2 de los 3 milestones no tienen ningún issue/HU asociados, intenta enfocarlos de forma que todos solucionen algún problema de los descritos en las HU.

No pasa nada *ahora*. Evidentemente, tendrán que tenerlo cuando se empiece a trabajar en ellos. En principio sólo debe tener HUs el milestone en el que se esté trabajando, el primero.

El resto; :+1: Gracias por ayudar.

pedrooot committed:
actualizo personas

pedrooot committed:
modifico persona Alberto

pedrooot committed:
completo milestones, historias y personas

pedrooot committed:
Update Milestones.md

pedromarting3 commented:
@jmramirezG he repasado lo que me dijiste, échale un vistazo cuando puedas. Muchas gracias!!

jmramirezG commented:
Hola Pedro!

Las historias de usuario creo que perfectas ahora!

Los milestones siguen avanzando!
Voy a proponerte un par de cambios más:

* En el [M1](https://github.com/pedromarting3/OnTime/milestone/1), me expresé mal en el comentario anterior cuando dije lo de *cada nodo corresponde a X y tiene un estado que indica Z, y cada arista corresponde a Y*. Al decir eso me refería a que explicitaras tú que era X, qué era Y y qué era Z, así se puede comprobar que el milestone está resuelto si se da esa condición.
* En el [M2](https://github.com/pedromarting3/OnTime/milestone/2) y [M3](https://github.com/pedromarting3/OnTime/milestone/3) explicita sin miedo las condiciones del archivo, es decir, qué condiciones tiene que cumplir la ruta para que sea válida para el hito? cuál es el formato del archivo que se genera?
Este punto quizás no sea tan urgente, pero eventualmente tendrás que explicitar en los milestones cuando se considerarán cumplidos, así que si ahora mismo no sabrías decir el formato del archivo, no te preocupes!!

Ya casi está, ánimo!!

pedrooot committed:
modifico milestones

Realmente no sé si el M3 está bien ya que no tengo clara todavía la extensión del archivo. He puesto .txt ya que si implemento ciertos operadores me puede ser útil.

pedrooot committed:
update errata

pedromarting3 commented:
Buenas @jmramirezG he solucionado lo que me comentaste, lo único que los [M2](https://github.com/pedromarting3/OnTime/milestone/2) y [M3](https://github.com/pedromarting3/OnTime/milestone/3) están menos completos que el [M1](https://github.com/pedromarting3/OnTime/milestone/1) porque pienso que dependen mucho del mismo. Una vez que implemente el M1, el resto cambiará.  
Cuando puedas le echas un vistazo, muchas gracias!

jmramirezG commented:
Buenas!

M1 y M2 los veo bien, al M3 quizás cambiarle el nombre para que se adapte más a lo que sería un PMV (modelo, método...) y diría que está bien, no sabria qué más decirte.

Tal vez @JJ pueda echarle un ojo y ser más claro con los posibles fallos :)

Ánimo! Que ya casi está.

JJ commented:
> Buenas!
> 
> M1 y M2 los veo bien, al M3 quizás cambiarle el nombre para que se adapte más a lo que sería un PMV (modelo, método...) y diría que está bien, no sabria qué más decirte.
> 
> Tal vez @JJ pueda echarle un ojo y ser más claro con los posibles fallos :)
> 
> Ánimo! Que ya casi está.

Los modelos y métodos tienen que estar empaquetados en un producto; un modelo solo no es un producto (en general)

JJ commented on docs/Milestones.md:
```diff
@@ -0,0 +1,20 @@
+    - _Componente Y -> contiene un valor que determina el nivel de congestión del nodo._  
+    - _Componente Z -> contiene un valor que nos dice si el nodo es transitable o no (no se considerará transitable si está en obras, hay un árbol en la carretera...)_  
+
+- [ ] [[M2] Método que genera la ruta dado un origen y el destino](https://github.com/pedromarting3/OnTime/milestone/2)  
```
Un método no es un producto. Una vez que tienes las estructuras de datos, tienes que completar un módulo que implemente parte de la estructura de negocio.

JJ commented on docs/Milestones.md:
```diff
@@ -0,0 +1,20 @@
+
+    _Este milestone se superará cuando nuestro servicio genere la ruta mas corta (en términos temporales) a partir de los dos datos suministrados. La ruta debe ser totalmente transitable y se calculará usando el modelo de la ciudad completo._
+
+- [ ] [[M3] Formato en el que nuestro servicio genera la ruta obtenida](https://github.com/pedromarting3/OnTime/milestone/3)  
```
Como te ha comentado @jmramirezG , tienes que formularlo como si fuera un producto.

pedrooot replied:
Perfecto, lo expresé como si fuese un método por lo que no cumplía con los requisitos del milestone.

JJ reviewed (changes requested):
Recuerda también que todo producto tiene que englobar los anteriores.

pedrooot committed:
modifico m2 y m3

pedromarting3 commented:
Hola!! @jmramirezG  

He formulado de nuevo los títulos (que expresaban los milestones como métodos y no como productos) pero pienso que ya deberían cumplir los requisitos.   

Si puedes revisarlo me sería de gran ayuda, perdón por las molestias 😵‍💫

jmramirezG reviewed (approved):
Diría que todo bien, a mi el uso de "generación" me parece correcto :)

pedromarting3 commented:
> Diría que todo bien, a mi el uso de "generación" me parece correcto :)

Perfecto, en cuanto @JJ diga que está bien mergeo

JJ commented on docs/Milestones.md:
```diff
@@ -0,0 +1,20 @@
+
+# Milestones  
+- [ ] [[M1] Estructura de datos para almacenar la información de la ciudad](https://github.com/pedromarting3/OnTime/milestone/1)  
```
Como comenté en clase, "estructura de datos" no es un entregable. Tendrás que decir de forma precisa qué es lo que vas a entregar. Una "estructura de datos" puede ser que la persona que programa te cuente qué va a usar: "un array" y listos.
La precisión es imprescindible en ingeniería

pedrooot replied:
Perfecto, ya he especificado que voy a usar un árbol.

JJ replied:
Agh, no. No son detalles de implementación, sino de "empaquetamiento"

pedrooot replied:
Decir que voy a usar un árbol es un detalle de implementación, pero si digo que usaré un array, ¿es un detalle de empaquetamiento?

JJ replied:
No, los detalles de implementación son los detalles de implementación. Y los detalles de empaquetamiento son qué es lo que le vas a entregar al product manager. ¿Le vas a entregar un array?

pedrooot replied:
Ahh vale, entiendo. Lo que le entregaría al product manager sería un modelo de la ciudad (si se implementa en forma de árbol por ejemplo ya es parte de la persona que lo programe, no es necesario especificarlo)

JJ replied:
No, es un producto que implemente el modelo. Y en clase os dije (varias veces) qué sería en este caso.

JJ commented on docs/Milestones.md:
```diff
@@ -0,0 +1,20 @@
+- [ ] [[M1] Estructura de datos para almacenar la información de la ciudad](https://github.com/pedromarting3/OnTime/milestone/1)  
+    _Según lo que hemos hablado con nuestros clientes, necesitaremos una estructura de datos que les permita obtener el recorrido por la ciudad de la formas más eficiente posible. Esta estructura nos permitirá crear un modelo de la ciudad completa._
+
+    _Este milestone se considerará superado cuando podamos representar una ciudad en forma de árbol. Cada nodo tendrá 3 componentes:_
```
Un milestone no tiene que dar detalles de implementación. La implementación la decide la persona que lo lleva a cabo.

pedrooot replied:
Elimino los detalles sobre el árbol.

JJ commented on docs/Milestones.md:
```diff
@@ -0,0 +1,20 @@
+    - _Componente Y -> contiene un valor que determina el nivel de congestión del nodo._  
+    - _Componente Z -> contiene un valor que nos dice si el nodo es transitable o no (no se considerará transitable si está en obras, hay un árbol en la carretera...)_  
+
+- [ ] [[M2] Generación de la ruta dado un origen y el destino](https://github.com/pedromarting3/OnTime/milestone/2)  
```
De la misma forma, tienes que decir "qué" lo va generar.

pedrooot replied:
Lo generaría el servicio a partir del árbol.

JJ replied:
Un milestone describe un producto. Por favor, trata de ceñirte a esa definición. Un "servicio" no es un producto. Tendrás que decir qué es en concreto ese servicio en términos que puedan ser entendidos por el equipo de programación y pueda responder a la pregunta "Qué hago ahora".

JJ commented on docs/Milestones.md:
```diff
@@ -0,0 +1,20 @@
+
+    _Este milestone se superará cuando nuestro servicio genere la ruta mas corta (en términos temporales) a partir de los dos datos suministrados. La ruta debe ser totalmente transitable y se calculará usando el modelo de la ciudad completo._
+
+- [ ] [[M3] Generación de un fichero legible en el que se plasma la ruta obtenida por el servicio](https://github.com/pedromarting3/OnTime/milestone/3)  
```
Posiblemente sea mejor que en vez de esto tengas un API REST, pero tampoco es imprescindible ahora.

JJ reviewed (changes requested):
Las HUs y la temporización de los milestones están bien, pero hay que ser más preciso diciendo exactamente qué es lo que se va a entregar.
:+1: buen trabajo por parte de los dos.

pedrooot committed:
Elimino descripción de implementación del árbol.

pedromarting3 commented:
Los cambios sugeridos por parte de @JJ han sido realizados, esperando su revisión 😺

JJ commented on docs/Milestones.md:
```diff
@@ -0,0 +1,16 @@
+
+# Milestones  
+- [ ] [[M1] Creación de un árbol utilizado para almacenar la información de la ciudad](https://github.com/pedromarting3/OnTime/milestone/1)  
```
Esto no describe un producto. Por favor, revisa el guión y la presentación que se hico al respecto.

JJ reviewed (changes requested):
Las HUs son correctas, pero todavía necesitas precisar qué productos vas a entregar en cada milestone. También están más o menos bien ordenados estos.

pedrooot committed:
Update m1 y m2

pedrooot committed:
Update M1 and M2

pedrooot committed:
Expreso milestones como productos,

pedromarting3 commented:
@JJ listo para revisión

pedrooot committed:
Update M3

pedrooot committed:
Update Milestones.md

pedromarting3 commented:
@JJ modifico m3, en la descripción especifico el formato del documento, que es un detalle de implementación. Lo he dejado porque en anteriores revisiones no comentamos nada al respecto.

JJ reviewed (approved):
👍
